kernel/src/fs/inode_handle/mod.rs

```diff
@@ -138,6 +138,9 @@ impl InodeHandle_ {
     pub fn set_status_flags(&self, new_status_flags: StatusFlags) {
         self.status_flags
             .store(new_status_flags.bits(), Ordering::Relaxed);
+        if let Some(file_io) = &self.file_io {
+            file_io.set_status_flags(new_status_flags);
```

Comment:
Do not duplicate atomic variables without proper locks. This is clearly wrong because it's racy:
```
[Thread A]                    [Thread B]

self.status_flags.store(A)
                              self.status_flags.store(B)
                              self.file_io.set_status(B)
sef.file_io.set_status(A)